includes/class-convertkit-output-restrict-content.php

```diff
@@ -360,14 +350,30 @@ public function maybe_run_subscriber_verification() {
 			return;
 		}
 
+		// If a nonce was specified, validate it now.
+		// It won't be provided if clicking the link in the magic link email.
+		if ( array_key_exists( '_wpnonce', $_REQUEST ) ) {
+			if ( ! wp_verify_nonce( sanitize_key( $_REQUEST['_wpnonce'] ), 'convertkit_restrict_content_subscriber_code' ) ) {
```

Comment:
I'm curious, if a nonce is optional, why do it at all?

Reply:
WordPress' coding standards recommends it for all form submissions. We widely do this for all other form submissions, so I think it's good practice to do this wherever possible (it's just this specific use case where we might not always have a nonce).